dai_cadcad/flip.py

```diff
@@ -0,0 +1,108 @@
+""" Flip Module
+    Class-based representation of the Flip smart contract
+    (contains only what is necessary for the simulation)
+"""
+
+
+class Flip:
+    ADDRESS = "flip"
+
+    vat = None
+    beg = None
+    ttl = None
+    tau = None
+    kicks = None
+    bids = {}
+
+    def __init__(self, vat, beg, ttl, tau, kicks, bids):
+        """"""
+
+        self.vat = vat
+
+        self.beg = beg
+        self.ttl = ttl
+        self.tau = tau
+        self.kicks = kicks
+        self.bids = bids
+
+    def flipper_kick(flipper, vat, user_id, gal, tab, lot, bid, end):
+        """Kicks off a new Flip auction."""
+
+        flipper["kicks"] += 1
+        bid_id = flipper["kicks"]
+
+        flipper["bids"][bid_id] = {
+            "bid": bid,
+            "lot": lot,
+            "guy": "cat",
+            "tic": 0,
+            "end": end,
+            "usr": user_id,
+            "gal": gal,
+            "tab": tab,
+        }
+
+        vat_flux(vat, flipper["ilk"], "cat", "flipper_eth", lot)
+
+    def flipper_tend(flipper, vat, bid_id, user_id, lot, bid, now):
+        """Places a tend bid on a Flipper auction."""
+
+        curr_bid = flipper["bids"][bid_id]
+
+        assert (
```

Comment:
We should replace the `asserts` everywhere with `require`, which you can import from `dai_cadcad.util`